### PR TITLE
[Model evaluation] Update GPU Support and add fallback to CPU logic

### PR DESCRIPTION
**Pull request by ankushbhatia2**

### CODE REVIEW

File: training/model_evaluation/src/task_factory/tabular/classification.py [hunk 1]
```diff
@@ -27,7 +30,7 @@ def _alternate_predict(self, X_test, **kwargs):
         Returns:
             _type_: _description_
         """
-        device = kwargs.get("device", "cpu")
+        device = kwargs.get("device", -1)
```
Comment (ShipraJain01) [inline training/model_evaluation/src/task_factory/tabular/classification.py:33]: For Tabular Classifier we want to retain Default Compute to -1? How about other Classifiers (Text / Image)?

**Review by ShipraJain01 — APPROVED** [review]
Approved with Comments.

File: training/model_evaluation/src/evaluate_model.py [hunk 2]
```diff
@@ -227,7 +223,7 @@ def test_model():
     parser.add_argument("--config-file-name", dest="config_file_name", required=False, type=str, default=None)
     parser.add_argument("--output", type=str, dest="output")
     parser.add_argument("--device", type=str, required=False, default="cpu", dest="device")
-    parser.add_argument("--batch-size", type=int, required=False, default=None, dest="batch_size")
+    parser.add_argument("--batch-size", type=int, required=False, default=1, dest="batch_size")
```
[the remark this answers is not in the document]
Comment (ankushbhatia2) [reply training/model_evaluation/src/evaluate_model.py:226]: For evaluate_model we anyway don't use the batch_size param. It was added by accident

File: training/model_evaluation/src/task_factory/tabular/classification.py (hunk 1 above)
Comment (ankushbhatia2) [reply training/model_evaluation/src/task_factory/tabular/classification.py:33]: "cpu" and "-1" are same. They both represent cpu in torch.

File: training/model_evaluation/src/evaluate_model.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (ankushbhatia2) [reply training/model_evaluation/src/evaluate_model.py:226]: This was actually added by Vision team for their tasks

**Pull request merged**